proxy/src/main/java/com/velocitypowered/proxy/protocol/netty/GS4QueryHandler.java

```diff
@@ -97,27 +106,43 @@ protected void channelRead0(ChannelHandlerContext ctx, DatagramPacket msg) throw
                         throw new IllegalStateException("Invalid query packet");
                     }
 
+                    boolean isBasic = queryMessage.readableBytes() == 0;
+                    QueryEvent event = new QueryEvent(isBasic ? QueryEvent.QueryType.BASIC : QueryEvent.QueryType.FULL,
+                            senderAddress, ComponentSerializers.PLAIN.serialize(server.getConfiguration().getMotdComponent()),
+                            ProtocolConstants.SUPPORTED_GENERIC_VERSION_STRING,
+                            server.getConfiguration().getQueryMap(), server.getPlayerCount(), server.getConfiguration().getShowMaxPlayers(),
+                            server.getConfiguration().getBind().getPort(), server.getConfiguration().getBind().getHostString(),
+                            server.getAllPlayers().stream().map(Player::getUsername).collect(Collectors.toList()),
+                            "Velocity",
+                            server.getConfiguration().shouldQueryShowPlugins() ? getRealPluginInformation() : Collections.emptyList()
+                    );
+                    try {
+                        server.getEventManager().fire(event).get();
```

Comment:
Never _ever_ use the blocking `get()` call. Instead prefer `whenCompleteAsync()` and use `ctx.channel().eventLoop()` as executor.